s3cortxutils/s3msgbus/s3msgbus/cortx_s3_msgbus.py

```diff
@@ -85,10 +85,15 @@ def setup_consumer(self, cons_id, group, msg_type, auto_ack, offset):
             return False, msg
         return True, None
 
-    def receive(self):
+    def receive(self, daemon_mode):
         """Receive the incoming message."""
         try:
-            message = self._consumer.receive()
+            if daemon_mode:
+                #timeout = 0.5 sec, by default, which is non-blocking
+                message = self._consumer.receive()
```

Comment:
if its running in daemon mode it should make a blocking call, else should make a non blocking call

Reply:
done